src/sc_client/core/async_sc_client_.py

```diff
@@ -0,0 +1,372 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import Awaitable, Callable, get_origin
+
+from sc_client.constants import common
+from sc_client.constants.common import CommandTypes, RequestType
+from sc_client.core.async_sc_connection import AsyncScConnection
+from sc_client.models import (
+    AsyncScEvent,
+    AsyncScEventParams,
+    Response,
+    ScAddr,
+    ScConstruction,
+    ScIdtfResolveParams,
+    ScLinkContent,
+    ScLinkContentData,
+    ScLinkContentType,
+    SCs,
+    SCsText,
+    ScTemplate,
+    ScTemplateParams,
+    ScTemplateResult,
+    ScTemplateValue,
+    ScType,
+)
+from sc_client.sc_exceptions import ErrorNotes, InvalidTypeError, ScServerError
+
+
+# pylint: disable=too-many-public-methods
+class AsyncScClient:
+    def __init__(self) -> None:
+        self._logger = logging.getLogger(f"{__name__}.{self.__class__.__name__}")
+        self._sc_connection = AsyncScConnection()
+
+    async def connect(self, url: str) -> None:
+        await self._sc_connection.connect(url)
+
+    def is_connected(self) -> bool:
+        return self._sc_connection.is_connected()
+
+    async def disconnect(self) -> None:
+        await self._sc_connection.disconnect()
+
+    def set_on_open_handler(self, on_open: Callable[[], Awaitable[None]]) -> None:
+        self._sc_connection.on_open = on_open
+
+    def set_on_close_handler(self, on_close: Callable[[], Awaitable[None]]) -> None:
+        self._sc_connection.on_close = on_close
+
+    def set_on_error_handler(self, on_error: Callable[[Exception], Awaitable[None]]) -> None:
+        self._sc_connection.on_error = on_error
+
+    def set_on_reconnect_handler(self, on_reconnect: Callable[[], Awaitable[None]]) -> None:
+        self._sc_connection.on_reconnect = on_reconnect
+
+    def set_reconnect_settings(self, retries: int = None, retry_delay: float = None) -> None:
+        self._sc_connection.reconnect_retries = retries or self._sc_connection.reconnect_retries
+        self._sc_connection.reconnect_retry_delay = retry_delay or self._sc_connection.reconnect_retry_delay
+
+    async def check_elements(self, *addrs: ScAddr) -> list[ScType]:
+        if not all(isinstance(addr, ScAddr) for addr in addrs):
+            raise InvalidTypeError(ErrorNotes.EXPECTED_OBJECT_TYPES_SC_ADDR)
+        if not addrs:
+            self._logger.warning("check_elements: empty params")
+            return []
+        payload = [addr.value for addr in addrs]
+        response = await self._send_message(RequestType.CHECK_ELEMENTS, payload)
+        data = [ScType(type_value) for type_value in response.payload]
+        return data
+
+    async def create_elements(self, constr: ScConstruction) -> list[ScAddr]:
+        if not isinstance(constr, ScConstruction):
+            raise InvalidTypeError(ErrorNotes.EXPECTED_OBJECT_TYPES, "ScConstruction")
+        if not constr.commands:
+            self._logger.warning("create_elements: empty ScConstruction")
```

Comment:
I think that in such cases warnings are no needed

Reply:
Yeah, you're right